W-20727606-retrieve-asset-metadata-using-MCP-server-endpoint-KS (#349)
mulesoft/docs-exchange: modules/ROOT/pages/retrieve-asset-metadata-using-MCP.adoc

> = Retrieve Asset Metadata Using an MCP Server Endpoint

Reviewer: ```suggestion
= Retrieving Asset Metadata Using an MCP Server Endpoint
```

ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:imagesdir: ../assets/images/

> If you don't have Flex Gateway deployed, use the Exchange MCP server endpoint to get asset and instance metadata from Exchange for your AI agents. The MCP server endpoint uses the Exchange API to search and retrieve this metadata programmatically.

Reviewer: ```suggestion
If you don’t have Flex Gateway deployed, use an MCP server endpoint to retrieve Agent Fabric asset and instance metadata from Exchange for your AI agents. The endpoint uses the Exchange API to search for and return this metadata.
```

== Using the MCP Server Endpoint

> Use this endpoint when configuring an MCP server connection in your AI client: `https://anypoint.mulesoft.com/exchange/mcp`

Reviewer: ```suggestion
Use this endpoint to configure an MCP server connection in your AI client: `https://anypoint.mulesoft.com/exchange/mcp`
```

You must authenticate to access private Exchange assets. Configure your AI client to use your Anypoint Platform credentials when connecting to the endpoint.

> == What You Can Do

Reviewer: ```suggestion

```

> After you connect your AI client to the Exchange MCP server, you can perform these tasks:

Reviewer: ```suggestion
After you connect your AI client to the MCP server endpoint, you can perform these tasks:
```

* *Search for assets* – Search by asset name, description, or keywords to find agents, MCP servers, large language models (LLMs), and other assets in Exchange.
* *Retrieve asset metadata* – Get metadata for an asset, including available versions, dependencies, and documentation so you can evaluate and use the asset in your projects.
* *Access reviews and ratings* – Read reviews and ratings for assets to compare options and choose the right asset for your use case.
* *Query user profile and permissions* – Check your user profile and permissions so you know what you can access, publish, or manage in Exchange.

== See Also

* xref:about-platform-apis.adoc[]

> * https://anypoint.mulesoft.com/exchange/portals/anypoint-platform/f1e97bc6-315a-4490-82a7-23abe036327a.anypoint-platform/exchange-experience-api/minor/2.4/pages/p1k-jaq/MCP%20Server/[Exchange MCP Server]

Reviewer: ```suggestion
* https://anypoint.mulesoft.com/exchange/portals/anypoint-platform/f1e97bc6-315a-4490-82a7-23abe036327a.anypoint-platform/exchange-experience-api/minor/2.4/pages/c05-go2/MCP%20Server/[MCP Server]
```
